File: python/intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        return list(set(nums1) & set(nums2))
# Alternate way using manual check (less optimal-gave 0ms runtime beating 100% of submissions but uses 12.64 MB memory beating 24.40% of submissions):
# class Solution(object):
#     def intersection(self, nums1, nums2):
#         kept = set(nums1)
#         intersections = set()
#         for num in nums2:
#             if num in kept:
#                 intersections.add(num)
#         return list(intersections)

File: java/intersection-of-two-arrays.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        Set<Integer> intersections = new HashSet<>();

        for (int num : nums1) {
            set.add(num);
        }

        for (int num : nums2) {
            if (set.contains(num)) {
                intersections.add(num);
            }
        }

        int[] result = new int[intersections.size()];
        int i = 0;
        for (int num : intersections) {
            result[i++] = num;
        }

        return result;
    }
}
